### ISSUE TITLE
SSR package breaks my code because it monkey patches Object.defineProperty.

### ISSUE THREAD

**Issue by trusktr** [skatejs/skatejs#1428]
I've got some code,

```js
Object.defineProperty(NewClass, 'prototype', {
  value: Ctor,
  writable: false,
})
```

but it fails because `@skatejs/ssr` overrides `Object.defineProperty`:

https://github.com/skatejs/skatejs/blob/4e64a4a5fd115141170befe84e5f35308c30e31b/packages/ssr/register/index.js#L3-L6

It'd be great if it didn't do that.

This is happening automatically because I'm using https://github.com/skatejs/skatejs/blob/master/packages/ssr/jest.js for my Jest test environment.

Why does SSR need this hack?

**Comment by trusktr** [comment]
By the way, if a descriptor is already `configurable:false` but `writable` is still `true`, then the descriptor is still allowed to be configured as long as the only thing being modified is `writable` being changed to `false`. This patch makes that feature fail in my case.

**Comment by trusktr** [comment]
> ```js
> // We patch this so Undom can't make any unconfigurable props.
> ```

We can't guarantee that Undom is the only thing calling `Object.defineProperty`. It's a normal JS API that anyone can call.

**Comment by trusktr** [comment]
PR: https://github.com/skatejs/skatejs/pull/1429

**Comment by treshugart** [comment]
Merged and released in 0.19.2 https://unpkg.com/@skatejs/ssr@0.19.2/register/index.js.

**Comment by trusktr** [comment]
Sweet thanks!
